src/js/components/Video.js

```diff
@@ -21,24 +15,23 @@ export default class Video extends Component {
   constructor () {
     super();
 
-    this._onResponsive = this._onResponsive.bind(this);
-    this._onPlaying = this._onPlaying.bind(this);
-    this._onPause = this._onPause.bind(this);
-    this._onEnded = this._onEnded.bind(this);
-    this._onClickControl = this._onClickControl.bind(this);
+    this._hasPlayed = false;
+    this._play = this._play.bind(this);
+    this._pause = this._pause.bind(this);
+    this._togglePlay = this._togglePlay.bind(this);
+    this._toggleMute = this._toggleMute.bind(this);
+    this._seek = this._seek.bind(this);
+    this._mute = this._mute.bind(this);
+    this._unmute = this._unmute.bind(this);
+    this._fullscreen = this._fullscreen.bind(this);
     this._onMouseMove = this._onMouseMove.bind(this);
-    this._onClickChapter = this._onClickChapter.bind(this);
-    this._onFullScreen = this._onFullScreen.bind(this);
 
-    this.state = { playing: false, progress: 0, iconSize: 'large' };
+    this.state = {};
   }
 
-  componentDidMount () {
-    this._responsive = Responsive.start(this._onResponsive);
-    let video = this.refs.video;
-    video.addEventListener('playing', this._onPlaying);
-    video.addEventListener('pause', this._onPause);
-    video.addEventListener('ended', this._onEnded);
+  componentWillMount () {
+    this._update = throttle(this._update.bind(this), 100).bind(this);
```

Comment:
I definitely understand your concerns. Generically speaking, I try to avoid using thridy-party libraries if I can solve the problem in a few lines of code. It feels to me that throttle is an example of that. 

It is not that we are trying to avoid using lodash, but every time we add a new dependency we need to check the license and run that with the open source team in HPE to make sure it is ok to use it. We can definitely do that, but I would say it would be easier/faster to implement the throttle on my own.

Hopefully that makes sense to you.